fuzz/zip_entry_read_fuzzer.c

```diff
@@ -0,0 +1,33 @@
+#include "zip.h"
+#include <stdint.h>
+#include <stdlib.h>
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, const size_t size) {
+  void *buf = NULL;
+  size_t bufsize = 0;
+
+  struct zip_t *zip = zip_stream_open((const char *)data, size, 0, 'r');
+  if (NULL == zip) {
+    goto end;
+  }
+
+  const ssize_t zip_entries_count = zip_entries_total(zip);
```

Comment:
Does it make sense?
If I understand correctly, we pass _random data_ as _zip stream_ and we try to open it for reading by `zip_stream_open`. 
It's not gonna work.

Reply:
@capuanob can you take a look into this fuzz (https://github.com/kuba--/zip/actions/runs/13177178831/job/36780519644)
The job fails with OOM error.

Reply:
@kuba-- In response to the original question, the data is not entirely random. The library is compiled with instrumentation that guides the fuzzer to slowly "learn" the grammar of a proper zip stream. After hundreds of thousands of iterations, it eventually begins producing semi-valid to valid zip streams that allows it to get deeper into the code.

Reply:
Yeah, that's great, why it OOM (https://github.com/kuba--/zip/actions/runs/13177178831/job/36780519644)
If it generates in memory stream we should set up some limits. Otherwise it does not make sense - it ran for 9 minutes and it failed with:
```
2025-02-06 11:28:42,995 - root - INFO - crash_type: out-of-memory.
2025-02-06 11:28:42,995 - root - INFO - crash_type: no-crashes.
2025-02-06 11:28:42,995 - root - INFO - Rule index: 0.
2025-02-06 11:28:42,995 - root - INFO - Not uploading crashes because on OSS-Fuzz.
2025-02-06 11:28:42,995 - root - INFO - Bug found.
```
nothing else.